csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -825,6 +999,87 @@ private void TestReusingDisposedRunOutput()
+                using (FixedBufferOnnxValue a = FixedBufferOnnxValue.CreateFromTensor(tensorA),
+                                            b = FixedBufferOnnxValue.CreateFromTensor(tensorB),
+                                            c = FixedBufferOnnxValue.CreateFromTensor(tensorC))
+                {
+                    session.Run(new[] { "input" }, new[] { a }, new[] { "output" }, new[] { b });
+                    session.Run(new[] { "input" }, new[] { b }, new[] { "output" }, new[] { c });
+                }
+
+                Assert.True(tensorC.SequenceEqual(tensorA));
+            }
+        }
+
+        [Fact]
+        private void TestReusingFixedBufferOnnxValueStringType()
+        {
+            // model takes 1x5 input of fixed type, echoes back
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_STRING.pb");
+            using (var session = new InferenceSession(modelPath))
+            {
+                var tensorIn = new DenseTensor<string>(new string[] { "a", "b", "c", "d", "e" }, new int[] { 1, 5 });
+                var tensorOut = new DenseTensor<string>(new string[] { "", "", "", "", "" }, new int[] { 1, 5 });
+                var ovIn = FixedBufferOnnxValue.CreateFromTensor(tensorIn);
+                var ovOut = FixedBufferOnnxValue.CreateFromTensor(tensorOut);
+
+                Assert.Throws<ArgumentException>("outputValues", () =>
+                {
+                    session.Run(new[] { "input" }, new[] { ovIn }, new[] { "output" }, new[] { ovOut });
+                });
+            }
+        }
+
+        [Fact]
+        private void TestReusingFixedBufferOnnxValueNonStringTypeMultiInferences()
+        {
+            // model takes 1x5 input of fixed type, echoes back
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_INT32.pb");
+            using (var session = new InferenceSession(modelPath))
+            {
+                var bufferInput = new int[5];
+                var bufferOutput = new int[5];
+                var tensorInput = new DenseTensor<int>(bufferInput, new int[] { 1, 5 });
+                var tensorOutput = new DenseTensor<int>(bufferOutput, new int[] { 1, 5 });
+
+                using (FixedBufferOnnxValue valueInput = FixedBufferOnnxValue.CreateFromTensor(tensorInput),
+                                            valueOutput = FixedBufferOnnxValue.CreateFromTensor(tensorOutput))
+                {
+                    var inputNames = new[] { "input" };
+                    var outputNames = new[] { "output" };
+                    var inputValues = new[] { valueInput };
+                    var outputValues = new[] { valueOutput };
+
+                    // run the model for multiple times
+                    for (var i = 0; i < 1000; i++)
+                    {
+                        // feed inputs
+                        var inputs = new int[] { 1, 2, 3, 4, 5 };
```

Comment:
Also, in case of string type (variable lengths between runs 1..1000), is there any way to throw friendly exception if that scenario is not handled? If so, can we add a test case to test for the exception string message? 

If there is no way to detect that scenario, and a user tries it anyways, what does the user see (invalid output, or some other exception?)

Reply:
for output, pre-allocated string tensors will cause an `ArgumentException` thrown in `Run()`, this is already covered in the other test case